addon/mixins/oauth2-implicit-grant-callback-route-mixin.js

```diff
@@ -0,0 +1,102 @@
+import Ember from 'ember';
+
+const { inject: { service }, Mixin, testing, computed, getOwner } = Ember;
+
+/**
+  __This mixin is used in the callback route when using OAuth 2.0 Implicit
+  Grant authentication.__ It implements the
+  {{#chrossLink "OAuth2ImplicitGrantCallbackRouteMixin/activate:method"}}{{/crossLink}}
+  method that retrieves and processes authentication parameters, such as
+  `access_token`, from the hash parameters provided in the callback URL by
+  the authentication server. The parameters are then passed to the
+  {{#crossLink "OAuth2ImplicitGrantAuthenticator"}}{{/crossLink}}
+
+  @class OAuth2ImplicitGrantCallbackMixin
+  @module ember-simple-auth/mixins/ouath2-implicit-grant-callback-mixin
+  @extends Ember.Mixin
+  @public
+*/
+
+export default Mixin.create({
+  /**
+   The session service.
+
+   @property session
+   @readOnly
+   @type SessionService
+   @public
+   */
+  session: service('session'),
+
+  /**
+    The authenticator that should be used to authenticate the callback. This
+    must be a subclass of the
+    {{#crossLink "OAuth2ImplicitGrantAuthenticator"}}{{/crossLink}}
+    authenticator.
+
+    @property authenticator
+    @type String
+    @default null
+    @public
+  */
+  authenticator: null,
+
+  /**
+    Any error that potentially occurs during authentication will be stored in
+    this property.
+
+    @property error
+    @type String
+    @default null
+    @public
+  */
+  error: null,
+
+  /**
+    Passes the hash received with the redirection from the authentication
+    server to the
+    {{#crossLink "OAuth2ImplicitGrantAuthenticator"}}{{/crossLink}} and
+    authenticates the session with the authenticator.
+
+    @method activate
+    @public
+  */
+  activate() {
+    if (!testing && this.get('_isFastBoot')) {
+      return;
+    }
+
+    let authenticator = this.get('authenticator');
+
+    let hash = this._parseResponse(this._windowLocationHash());
+
+    this.get('session').authenticate(authenticator, hash).catch((err) => {
+      this.set('error', err);
+    });
+  },
+
+  _isFastBoot: computed(function() {
+    const fastboot = getOwner(this).lookup('service:fastboot');
+
+    return fastboot ? fastboot.get('isFastBoot') : false;
+  }),
+
+  _windowLocationHash() {
+    // we wrap this so we can stub it with sinon
+    return window.location.hash;
```

Comment:
That's what I was trying to do originally but it doesn't sound like that's possible (see below). It sounds like some of the window properties are set in such a way that they cannot be overridden. This seemed to be the most general solution, but I'm certainly open to other ideas if you have anything that you like better. It's definitely not the prettiest.  

References:
http://stackoverflow.com/questions/36678302/stubbing-window-location-href-with-sinon
http://stackoverflow.com/a/11960524/977044
https://groups.google.com/forum/?fromgroups=#!topic/sinonjs/MMYrwKIZNUU%5B1-25%5D